gnucash/gnome/gnc-plugin-budget.c

```diff
@@ -233,6 +241,24 @@ gnc_plugin_budget_cmd_copy_budget (GtkAction *action,
         gnc_plugin_budget_cmd_new_budget (action, user_data);
 }
 
+/* user selects budget to delete */
+static void
+gnc_plugin_budget_cmd_delete_budget (GtkAction *action,
+                                     GncMainWindowActionData *user_data)
+{
+    QofBook *book = gnc_get_current_book ();
+    QofCollection *col = qof_book_get_collection (book, GNC_ID_BUDGET);
+    GncBudget *bgt;
+
+    if (qof_collection_count (col) == 0) return;
+
+    g_return_if_fail (user_data != NULL);
```

Comment:
The return test I think should be below GncBudget *bgt;
and move the get collection to be below the g_return_if_fail, if you are testing a parameter, there is no point in doing any thing unless it passes and you could even do this ...

    if (qof_collection_count (qof_book_get_collection (book, GNC_ID_BUDGET)) == 0)
        return;